tests/integration/data/DatabaseTest.php

```diff
@@ -483,6 +483,33 @@ public function testSubqueryWithMultipleRecords() {
 		));
 		$this->assertEqual(2, $results->count());
 	}
+
+	public function testUpdateWithMultiThread() {
```

Comment:
no it isn't parallelism, just the name is multiThread .

in this test case we have two different object that affect on each other.